### PR TITLE
Tests/test tools test contract

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

```
Name                              Stmts   Miss  Cover   Missing
---------------------------------------------------------------
aea/test_tools/test_contract.py      85      0   100%
...
```

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`aea hash all` and `aea packages lock --check`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

**Review by 0xArdi — APPROVED** [review]

File: tests/test_test_tools/test_test_contract.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    """Consume the iterator"""
+    collections.deque(iterator, maxlen=0)
+
+
+@contextmanager
+def as_context(*contexts: Any) -> Generator[None, None, None]:
+    """Set contexts"""
+    with ExitStack() as stack:
+        consume(map(stack.enter_context, contexts))
+        yield
```
[the remark this answers is not in the document]
Comment (DavidMinarsch) [reply tests/test_test_tools/test_test_contract.py:71]: They look like useful tool in tests/utils. Do you think they'll be used outside these tests here? It's a tradeoff between larger library and reuse

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
For tests to pass change https://ipfs.io/ to https://ipfs.tech/ - seems they moved

File: tests/test_test_tools/test_test_contract.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Karrenbelt) [reply tests/test_test_tools/test_test_contract.py:71]: I would expect it to be used, I would have used it in other tests I wrote and I've also [used the pattern in open-autonomy](https://github.com/valory-xyz/open-autonomy/blob/7e4b52e4aa1590aa2663cb4b83be9fe8b733141b/packages/valory/skills/registration_abci/tests/test_behaviours.py#L80-L90) before

**Comment by Karrenbelt** [comment]
> For tests to pass change https://ipfs.io/ to https://ipfs.tech/ - seems they moved

I tried to address this on main but to no avail, see https://github.com/valory-xyz/open-aea/pull/385

**Review by angrybayblade — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**